src/test/java/teammates/logic/core/FeedbackResponsesLogicTest.java

```diff
@@ -1712,6 +1712,116 @@ public void testGetSessionResultsForUser_orphanResponseInDB_shouldStillHandleCor
+                .build();
+        boolean response = FeedbackResponsesLogic.inst().isResponseOfFeedbackQuestionVisibleToStudent(question);
+        assertEquals(response, true);
+    }
+
+    @Test
+    public void testResponseVisibleToStudentWhenReceiverIsOwnTeamMember() {
+        List<FeedbackParticipantType> showResponseTo = new ArrayList<>();
+        showResponseTo.add(FeedbackParticipantType.RECEIVER);
+
+        FeedbackQuestionAttributes question = FeedbackQuestionAttributes.builder()
+                .withShowResponsesTo(showResponseTo)
+                .withRecipientType(FeedbackParticipantType.OWN_TEAM_MEMBERS)
+                .withGiverType(FeedbackParticipantType.STUDENTS)
+                .build();
+        boolean response = FeedbackResponsesLogic.inst().isResponseOfFeedbackQuestionVisibleToStudent(question);
+        assertEquals(response, true);
+    }
+
+    @Test
+    public void testResponseVisibleToStudentWhenReceiverIsGiver() {
+        List<FeedbackParticipantType> showResponseTo = new ArrayList<>();
+        showResponseTo.add(FeedbackParticipantType.RECEIVER);
+
+        FeedbackQuestionAttributes question = FeedbackQuestionAttributes.builder()
+                .withShowResponsesTo(showResponseTo)
+                .withRecipientType(FeedbackParticipantType.GIVER)
+                .withGiverType(FeedbackParticipantType.STUDENTS)
+                .build();
+        boolean response = FeedbackResponsesLogic.inst().isResponseOfFeedbackQuestionVisibleToStudent(question);
+        assertEquals(response, true);
+    }
+
+    @Test
+    public void testResponseNotVisibleToStudentWhenReceiverTeamMembers() {
+        List<FeedbackParticipantType> showResponseTo = new ArrayList<>();
+        showResponseTo.add(FeedbackParticipantType.RECEIVER_TEAM_MEMBERS);
+
+        FeedbackQuestionAttributes question = FeedbackQuestionAttributes.builder()
+                .withShowResponsesTo(showResponseTo)
+                .withRecipientType(FeedbackParticipantType.NONE)
+                .withGiverType(FeedbackParticipantType.NONE)
+                .build();
+        boolean response = FeedbackResponsesLogic.inst().isResponseOfFeedbackQuestionVisibleToStudent(question);
+        assertEquals(response, false);
+    }
+
+    @Test
+    public void testResponseVisibleToStudentWhenShownToOwnTeamMembers() {
+        List<FeedbackParticipantType> showResponseTo = new ArrayList<>();
+        showResponseTo.add(FeedbackParticipantType.OWN_TEAM_MEMBERS);
+
+        FeedbackQuestionAttributes question = FeedbackQuestionAttributes.builder()
+                .withShowResponsesTo(showResponseTo)
+                .withRecipientType(FeedbackParticipantType.NONE)
+                .withGiverType(FeedbackParticipantType.TEAMS)
+                .build();
+        boolean response = FeedbackResponsesLogic.inst().isResponseOfFeedbackQuestionVisibleToStudent(question);
+        assertEquals(response, true);
+    }
```

Comment:
this test case is a positive test case for 
question.getGiverType() == FeedbackParticipantType.TEAMS
|| question.isResponseVisibleTo(FeedbackParticipantType.OWN_TEAM_MEMBERS, (lines 320 and 321 of FeedbackResponsesLogic)

given that it is an OR case, let's seperate the 2 conditions into their own test case